google/cloud/dataproc_spark_connect/session.py

```diff
@@ -314,6 +312,50 @@ def create_session_pbar():
                     session_response, dataproc_config.name
                 )
 
+        def _display_session_link_on_creation(self, session_id):
+            assert all(
+                [
+                    session_id is not None,
+                    self._region is not None,
+                    self._project_id is not None,
+                ]
+            )
+
+            session_url = f"https://console.cloud.google.com/dataproc/interactive/{self._region}/{session_id}?project={self._project_id}"
+
+            try:
+                from IPython.display import display, HTML
+                from IPython.core.interactiveshell import InteractiveShell
+
+                if not InteractiveShell.initialized():
+                    raise DataprocSparkConnectException(
+                        "Not in an Interactive IPython Environment"
+                    )
+                html_element = f"""
+                    <div>
+                        <p>Creating Dataproc Spark Session<p>
+                        <p><a href="{session_url}">Dataproc Session</a></p>
+                    </div>
+                """
+                display(HTML(html_element))
+            except (ImportError, DataprocSparkConnectException):
+                print(f"Creating Dataproc Session: {session_url}")
+
+        def _print_session_created_message(self):
+            message = f"Dataproc Session was successfully created"
+            try:
+                from IPython.display import display, HTML
+                from IPython.core.interactiveshell import InteractiveShell
+
+                if not InteractiveShell.initialized():
+                    raise DataprocSparkConnectException(
+                        "Not in an Interactive IPython Environment"
+                    )
+                html_element = f"<div><p>{message}</p></div>"
+                display(HTML(html_element))
+            except (ImportError, DataprocSparkConnectException):
+                print(message)
```

Comment:
There's a lot of duplication between this method and the one above it.

Let's factor out a helper method that displays either the rich HTML element or the plain text depending on whether or not the rich element is supported.